File: q10-q19/q14/q14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string solution = "";
        if (!strs.size())
            return solution;
        int shortest_str_length = INT_MAX;
        int list_size = strs.size();
        bool is_prefix = true;
        for(int i = 0; i < list_size; i++){
            if(strs[i].size() < shortest_str_length)
                shortest_str_length = strs[i].size();
        }
        for(int char_index = 0; char_index < shortest_str_length; char_index++){
            for(int str_index = 0; str_index < list_size - 1; str_index++){
                if(strs[str_index][char_index] != strs[str_index+1][char_index]){
                    is_prefix = false;
                    break;
                }
            }
            if (is_prefix){
                solution += strs[0][char_index];
            }
        }
        return solution;
    }
};

File: q10-q19/q14/q14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        solution = ""
        if len(strs) == 0:
            return solution;
        list_size = len(strs)
        shortest_str = inf
        is_prefix = True
        for i in range(list_size):
            if(len(strs[i]) < shortest_str):
                shortest_str = len(strs[i])
                
        for char_index in range(shortest_str):
            for str_index in range(list_size-1):
                if strs[str_index][char_index] != strs[str_index+1][char_index]:
                    is_prefix = False
                    break
            if is_prefix:
                solution += strs[0][char_index]
        return solution
